deployment-guide.md

```diff
@@ -19,98 +19,47 @@ You can deploy your contracts and run an end-to-end test or demo as follows:
-    export APP_ADDRESS="#copy app address from the deploy logs#"
+    forge script --rpc-url http://localhost:8545 --broadcast script/Deploy.s.sol
     ```
 
-4. Start the Bonsai Ethereum Relay by running:
-
+    This command should output something similar to:
+    
     ```bash
-    RISC0_DEV_MODE=true cargo run --bin bonsai-ethereum-relay-cli -- run --relay-address "$BONSAI_RELAY_ADDRESS"
+    ...
+    == Logs ==
+    Deployed RiscZeroGroth16Verifier to 0x5FbDB2315678afecb367f032d93F642f64180aa3
+    Deployed EvenNumber to 0xe7f1725E7734CE288F8367e1Bb143E90bb3F0512
+    ...
     ```
 
-    The relay will keep monitoring the chain for callback requests, generated when your contract calls `bonsaiRelay.requestCallback(...)`, and relay their result back to your contract after computing them.
-    Keep the relay running and switch to a new terminal.
-
-    Setting `RISC0_DEV_MODE=true` deploys the `BonsaiTestRelay`, for use in local development and testing, instead of the fully verifying `BonsaiRelay` contract.
-    See the section below on using the fully-verifying relay for more information on this setting and testnet deployment.
-
 ### Interact with your deployment:
 
-#### Off-chain callback request
-
-The Relay exposes an HTTP REST API interface that can be used to directly send *off-chain* callback requests to it.
-It also provides an SDK in Rust that can be used to interact with it. You can check out this [example](relay/examples/offchain_request.rs).
-
-1. Send a callback request directly to the Relay by running:
-
-    ```bash
-    BONSAI_API_KEY="YOUR_API_KEY_OR_EMPTY_IF_DEV_MODE" cargo run --example offchain_request "$APP_ADDRESS" 10
-    ```
-
-2. Check the relayed result:
-
+1. Query the state:
     ```bash
-    cast call "$APP_ADDRESS" 'fibonacci(uint256)' 10
+    cast call --rpc-url http://localhost:8545 0xe7f1725e7734ce288f8367e1bb143e90bb3f0512 'get()(uint256)'
     ```
 
-This example's arguments are the `BonsaiStarter` contract address and the number, N, to compute the Nth Fibonacci number.
-You may need to change these values accordingly.
-
-The Relay source code with its SDK can be found in the [risc0/risc0] github repo.
-
-#### On-chain callback request
-
-1. Send a transaction to the starter contract:
-
-    ```bash
-    cast send --private-key 0x59c6995e998f97a5a0044966f0945389dc9e86dae88c7a8412f4603b6b78690d --gas-limit 100000 "$APP_ADDRESS" 'calculateFibonacci(uint256)' 5
-    ```
-
-2. Check the relayed result:
+2. Publish a new state
+    > ***Note:*** *This requires having access to a Bonsai API Key. To request an API key [complete the form here](https://bonsai.xyz/apply).*
 
     ```bash
-    cast call "$APP_ADDRESS" 'fibonacci(uint256)' 5
+    RISC0_DEV_MODE=false BONSAI_API_URL="BONSAI_API_URL" BONSAI_API_KEY="BONSAI_API_KEY" cargo run --release -- publish \
+        --chain-id=31337 \
+        --rpc-url=http://localhost:8545 \
+        --contract=e7f1725E7734CE288F8367e1Bb143E90bb3F0512 \
+        --input=12345678
```

Comment:
How is this input handled? It seems like it'll need to be parse, and this parsing seems like it may be application specific, but it's unclear to me how this ends up happening.

Reply:
Currently the assumption is that the input is passed via cli as a string, then the user needs to implement the `GuestInterface` trait, specifically the function `serialize_input(input: String) -> Result<Vec<u8>>`